src/containers/student-home-page/find-tutor-block/FindTutorBlock.jsx

```diff
@@ -3,66 +3,57 @@ import { useNavigate } from 'react-router'
 import { useTranslation } from 'react-i18next'
 
 import TitleBlock from '~/components/title-block/TitleBlock'
-import Button from '@mui/material/Button'
-import IconButton from '@mui/material/IconButton'
-import TextField from '@mui/material/TextField'
 import SearchIcon from '@mui/icons-material/Search'
 
 import useBreakpoints from '~/hooks/use-breakpoints'
 import { styles } from '~/containers/student-home-page/find-tutor-block/find-tutor-block.styles'
 import bag from '~/assets/img/student-home/bag.png'
 import { translationKey } from '~/containers/student-home-page/find-tutor-block/constants'
 import { authRoutes } from '~/router/constants/authRoutes'
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
+import AppButton from '~/components/app-button/AppButton'
+import { Link } from 'react-router-dom'
 
 const FindTutorBlock = () => {
-  const [filter, setFilter] = useState('')
+  const [inputValue, setInputValue] = useState('')
   const { t } = useTranslation()
   const navigate = useNavigate()
   const { isMobile } = useBreakpoints()
 
   const onChange = (e) => {
-    setFilter(e.target.value)
+    setInputValue(e.target.value)
   }
 
-  const redirect = useCallback(() => {
-    return navigate(authRoutes.findOffers.path, { state: { filter } })
-  }, [filter, navigate])
-
   const handleEnterPress = useCallback(
     (e) => {
-      if (e.key === 'Enter' && filter) {
-        redirect()
+      if (e.key === 'Enter' && inputValue) {
+        navigate(authRoutes.findOffers.path, { state: { inputValue } })
       }
     },
-    [filter, redirect]
+    [inputValue]
```

Comment:
Why do you move it to useEffect and also you forget to add a variable to dependency?